app/controllers/offices_controller.rb

```diff
@@ -0,0 +1,24 @@
+class OfficesController < ApplicationController
+  def index
+    return unless search
+    return @error_message = search.error_message unless search.valid?
+
+    @result = search.result
```

Comment:
This `result` method on the two existing search classes (`ZipCodeSearch` and `CoordinatesSearch`) returns a hash with `latitude`, `longitude`, and (for ZIP searches) `zip_code`.

Reply:
Why is zip_code returned for zip code search results? Where is it used?

Reply:
`@result[:zip_code]` is used on line 66 of `app/views/offices/index.html.erb` to show a "you searched for"-style message:

```erb
<p>Displaying results near <b><%= @result[:zip_code] || "#{@result[:latitude]}, #{@result[:longitude]}" %></b><%= " (page #{@transportation_offices.current_page} of #{@transportation_offices.total_pages})" if @transportation_offices.total_pages.positive? %>.</p>
```

I suppose we _could_ change this to use `params[:postal_code]` and keep the two search class `result` methods similar. Not sure which is better, though.

Reply:
@aileendds Oh, what if we added a `query` (or similar) method to the search classes that returns what the user entered?

```
class CoordinatesSearch
  …
  
  def query
    search_params[:coordinates]
  end
```

That could then be exposed to the view via the controller by setting `@search` (or similar):

```erb
<p>Displaying results near <b><%= @search.query %></b>.</p>
```

What do you think? That'd then keep the `result` methods in the search classes returning similar hashes.

Reply:
Yes, that works!